File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>
using namespace std;

int N, num;
string s;

int main(void) {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    cin >> N;

    queue<int> q;
    while(N--) {
        cin >> s;
        
        if(s == "push") {
            cin >> num;
            q.push(num);
        }
        if(s == "pop") {
            if(q.empty()) cout << -1 << '\n';
            else {cout << q.front() << '\n'; q.pop();}
        }
        if(s == "front") {
            if(q.empty()) cout << -1 << '\n';
            else {cout << q.front() << '\n';}
        }
        if(s == "back") {
            if(q.empty()) cout << -1 << '\n';
            else {cout << q.back() << '\n';}
        }
        if(s == "size") {
            cout << q.size() << '\n';
        }
        if(s == "empty") {
            cout << q.empty() << '\n';
        }   
    }
}

File: 백준/Silver/10845. 큐/큐.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main(args: Array<String>) {

    val br = BufferedReader(InputStreamReader(System.`in`));
    val bw = BufferedWriter(OutputStreamWriter(System.out));

    val N = br.readLine().toInt();

    val queue = LinkedList<String>();
    (1..N).forEach() {
        val op = br.readLine().split(" ");

        if(op[0]=="push") {
            queue.offer(op[1]);
        }
        if(op[0]=="pop") {
            if(queue.isEmpty()) bw.write("-1\n");
            else {
                bw.write(queue.poll()+"\n");
            }
        }
        if(op[0]=="size") {
            bw.write(queue.size.toString() + "\n");
        }
        if(op[0]=="empty") {
            if(queue.isEmpty()) bw.write("1\n");
            else bw.write("0\n");
        }
        if(op[0]=="front") {
            if(queue.isEmpty()) bw.write("-1\n");
            else bw.write(queue.peekFirst()+"\n");
        }
        if(op[0]=="back") {
            if(queue.isEmpty()) bw.write("-1\n");
            else bw.write(queue.peekLast()+"\n");
        }
    }

    br.close();
    bw.flush();
    bw.close();
}